tools/src/wyvern/tools/errors/ErrorMessage.java

```diff
@@ -109,16 +109,20 @@
     NO_ABSTRACT_TYPES_IN_OBJECTS("Abstract types may not be declared in objects or modules, only in type definitions", 0),
     METHODS_MUST_BE_INVOKED("Cannot access a method as if it were a field; use ()", 0),
     TSL_ERROR("Error in type-specific language: %ARG", 1),
-    CANNOT_APPLY_TYPE_PARAMETERS("Cannot apply type parameters: type %ARG is abstract", 1),
-    NO_TYPE_MEMBER("Cannot find enough type members to apply type parameters", 0),
+    CANNOT_APPLY_GENERIC_ARGUMENTS("Cannot apply generic arguments: type %ARG is abstract", 1),
+    NO_TYPE_MEMBER("Cannot find enough type members to apply type argument %ARG", 1),
+    NO_EFFECT_MEMBER("Cannot find enough effect members to apply effect argument %ARG", 1),
+    NON_TYPE_ARGUMENT("Cannot apply generic argument of kind %ARG to an abstract type", 1),
+    NON_EFFECT_ARGUMENT("Cannot apply generic argument of kind %ARG to an abstract effect", 1),
 
     // effects errors
     MISTAKEN_DSL("Invalid characters for effect--should not be a DSL block: \"effect %ARG = {%ARG}\"", 2),
     UNDEFINED_EFFECT("Effect \"%ARG\" is undefined", 1),
     EFFECT_NOT_IN_SCOPE("Effect \"%ARG\" not found in scope", 1),
     //    EFFECT_OF_VAR_NOT_FOUND("Effect \"%ARG\" not found for variable \"%ARG\"", 2),
     RECURSIVE_EFFECT("Effect \"%ARG\" is being defined recursively", 1),
-    UNKNOWN_EFFECT("Effects of method call \"%ARG\" are unknown", 1);
+    UNKNOWN_EFFECT("Effects of method call \"%ARG\" are unknown", 1),
+    UNNAMED_EFFECT("Effect \"%ARG\" must be attached to a named object", 1);
```

Comment:
This error message currently seems not to be used. Please hold off on adding it until it is used.